source/tools/testConsistency/ClientApp/app.test.js

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. 
+// Licensed under the MIT License. 
+// To run these tests, first ensure that the flask app for nodeJs is running at 127.0.0.1 port 5000
+// and the flask app for .NET WPF is running at 127.0.0.1 port 4000. 
+const request = require("supertest");
+const app = require("./app");
+const fs = require("fs");
+
+test("The GET request to the index page should return a response code 200", function(done) {
+     request(app)
+      .get("/")
+      .then(function(response) {
+        expect(response.statusCode).toBe(200);
+        done();
+      });
+});
+
+test("Sending no card data should still return a response code 200", function(done) {
+    const templateBuffer = fs.readFileSync("../../../../samples/v1.0/Scenarios/ActivityUpdate.json");
+    const template = templateBuffer.toString('utf8')
+    request(app)
+      .post("/")
+      .send(cardTemplate=template, cardData="")
+      .then(function(response){
+          expect(response.statusCode).toBe(200);
+          done();
+      });
+});
+
+test("Sending no card template should still return a response code 200", function(done) {
+    const data = {
+                    "name": "Matt Hidinger"
+                 }
+    request(app)
+      .post("/")
+      .send(cardTemplate="", cardData=data)
+      .then(function(response){
+          expect(response.statusCode).toBe(200);
+          done();
+      });
+});
+
+test("Sending an empty form should still return a response code 200", function(done) {
+    const data = {
+                    "name": "Matt Hidinger"
+                 }
+    request(app)
+      .post("/")
+      .send(cardTemplate="", cardData="")
+      .then(function(response){
+          expect(response.statusCode).toBe(200);
+          done();
+      });
+});
+
+test("Sending invalid JSON data should still return a response code 200", function(done) {
```

Comment:
>d JSON data should still  [](start = 20, length = 25)

Interesting - so can i send in any old string here as well instead of only well-formed json? If we are not validating our parameters for being well-formed json then why are we defining our parameters as accepting json data when it really should be any string right? (I am guessing the reason is because we want to let the actual per-platform serialization and renderer stacks catch and report the invalid json errors rather than intervening ourselves and causing potential confusion?)

Reply:
That is correct. I want to ensure that in case of an invalid json string, all errors are handled by the renderers so that we can ensure consistent error handling as well